lib/routes-pub.js

```diff
@@ -7,19 +7,28 @@ const { TextDecoder, TextEncoder } = require('util')
 const dfns = require('date-fns')
 const isValidAccount = require('eos-name-verify')
 const request = require('request')
+const { getCompetition } = require('./gr-client')
 
 const {
   formatCandles,
   formatTrades,
   checkCandlesReq,
-  log
+  log,
+  sendGenericError
 } = require('./utils')
 
 class RoutesPub {
   constructor (conf, plugins) {
     this.conf = conf
     this.plugins = plugins
 
+    this.onTimeHttpRequest = this.onTimeHttpRequest.bind(this)
+    this.onTosHttpRequest = this.onTosHttpRequest.bind(this)
```

Comment:
why rebinding Tos handlers etc here?

Reply:
In order to omit `bind` call wherever you pass the method as a callback. I wanted to see what you think about that.